The job **`mingw-check`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/runs/8165083849?check_suite_focus=true) [(plain)](https://github.com/rust-lang/rust/commit/71b721f554fc3e988f67d54c3239b378aae88a77/checks/8165083849/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
    Checking alloc v0.0.0 (/checkout/library/alloc)
    Checking cfg-if v0.1.10
    Checking adler v0.2.3
    Checking rustc-demangle v0.1.21
error[E0520]: `is_zero` specializes an item from a parent `impl`, but that item is not marked `default`
   --> library/alloc/src/vec/is_zero.rs:170:5
    |
150 |   unsafe impl<T: IsZero> IsZero for Wrapping<T> {
    |   --------------------------------------------- parent `impl` is here
...
170 | /     fn is_zero(&self) -> bool {
171 | |         self.0 == 0
172 | |     }
    | |_____^ cannot specialize default item `is_zero`
    |
    = note: to specialize, `is_zero` in the parent `impl` must be marked `default`

error[E0520]: `is_zero` specializes an item from a parent `impl`, but that item is not marked `default`
   --> library/alloc/src/vec/is_zero.rs:177:5
    |
150 |   unsafe impl<T: IsZero> IsZero for Wrapping<T> {
    |   --------------------------------------------- parent `impl` is here
...
177 | /     fn is_zero(&self) -> bool {
178 | |         self.0 == 0
179 | |     }
    | |_____^ cannot specialize default item `is_zero`
    |
    = note: to specialize, `is_zero` in the parent `impl` must be marked `default`

error[E0520]: `is_zero` specializes an item from a parent `impl`, but that item is not marked `default`
   --> library/alloc/src/vec/is_zero.rs:184:5
    |
157 |   unsafe impl<T: IsZero> IsZero for Saturating<T> {
    |   ----------------------------------------------- parent `impl` is here
...
184 | /     fn is_zero(&self) -> bool {
185 | |         self.0 == 0
186 | |     }
    | |_____^ cannot specialize default item `is_zero`
    |
    = note: to specialize, `is_zero` in the parent `impl` must be marked `default`

error[E0520]: `is_zero` specializes an item from a parent `impl`, but that item is not marked `default`
   --> library/alloc/src/vec/is_zero.rs:191:5
    |
157 |   unsafe impl<T: IsZero> IsZero for Saturating<T> {
    |   ----------------------------------------------- parent `impl` is here
...
191 | /     fn is_zero(&self) -> bool {
192 | |         self.0 == 0
193 | |     }
    | |_____^ cannot specialize default item `is_zero`
    |
    = note: to specialize, `is_zero` in the parent `impl` must be marked `default`
For more information about this error, try `rustc --explain E0520`.
error: could not compile `alloc` due to 4 previous errors
Build completed unsuccessfully in 0:01:17
```

</details>

--- a/library/alloc/src/lib.rs
+++ b/library/alloc/src/lib.rs
@@ -134,6 +134,7 @@
 #![feature(ptr_metadata)]
 #![feature(ptr_sub_ptr)]
 #![feature(receiver_trait)]
+#![feature(saturating_int_impl)]
 #![feature(set_ptr_value)]
 #![feature(slice_from_ptr_range)]
 #![feature(slice_group_by)]

--- a/library/alloc/src/vec/is_zero.rs
+++ b/library/alloc/src/vec/is_zero.rs
@@ -1,3 +1,5 @@
+use core::num::{Saturating, Wrapping};
+
 use crate::boxed::Box;
 
 #[rustc_specialization_trait]
@@ -144,3 +146,17 @@ impl_is_zero_option_of_nonzero!(
     NonZeroUsize,
     NonZeroIsize,
 );
+
+unsafe impl<T: IsZero> IsZero for Wrapping<T> {
+    #[inline]
+    fn is_zero(&self) -> bool {
+        self.0.is_zero()
+    }
+}
+
+unsafe impl<T: IsZero> IsZero for Saturating<T> {
+    #[inline]
+    fn is_zero(&self) -> bool {
+        self.0.is_zero()
+    }
+}

